src/arithmetic/rewrite_simplify.cc

```diff
@@ -1127,6 +1131,8 @@ Mutate_(const Not* op, const Expr& self) {
   TVM_TRY_REWRITE(!(x > y), x <= y);
   TVM_TRY_REWRITE(!(x == y), x != y);
   TVM_TRY_REWRITE(!(x != y), x == y);
+  TVM_TRY_REWRITE(!(x || y), (!x) && (!y));
+  TVM_TRY_REWRITE(!(x && y), (!x) || (!y));
```

Comment:
It would be better to use `TVM_TRY_RECURSIVE_REWRITE` here, because otherwise `not`s are not propagated all the way down in some expressions (like `((max(max(i, x), y) <= 9) && ((x == i) || (y == i)))`).

Reply:
updated